pkg/operator/bootstrapteardown/bootstrap_teardown_controller.go

```diff
@@ -159,7 +168,34 @@ func (c *BootstrapTeardownController) removeBootstrap() error {
 		return nil
 	}
 
+	// check to see if bootstrapping is complete
+	bootstrapFinished := false
+	bootstrapFinishedConfigMap, err := c.kubeClient.CoreV1().ConfigMaps("kube-system").Get("bootstrap", metav1.GetOptions{})
+	if err != nil {
+		klog.V(2).Infof("non-fatal error getting configmap: %v", err)
+	} else {
+		if bootstrapFinishedConfigMap.Data["status"] == "complete" {
+			bootstrapFinished = true
+		}
+	}
+	if !bootstrapFinished {
+		// check to see if we have an event yet.
+		_, err := c.kubeClient.CoreV1().Events("kube-system").Get("bootstrap-finished", metav1.GetOptions{})
+		if err != nil {
+			klog.V(2).Infof("non-fatal error getting bootstrap-finished event: %v", err)
+		} else {
+			bootstrapFinished = true
+		}
+	}
+
+	if !bootstrapFinished {
+		c.eventRecorder.Event("DelayingBootstrapTeardown", "cluster-bootstrap is not yet finished")
+		return nil
+	}
```

Comment:
```suggestion
		c.eventRecorder.Event("DelayingBootstrapTeardown", "cluster-bootstrap is not yet finished")
		return nil
	}
}
```

Reply:
thanks! we will circle back around for cleanup. I appreciate the notes.